mmdet/models/dense_heads/cascade_rpn_head.py

```diff
@@ -0,0 +1,623 @@
+from __future__ import division
+
+import torch
+import torch.nn as nn
+from mmcv.cnn import normal_init
+from mmcv.ops import DeformConv2d
+
+from mmdet.core import (RegionAssigner, build_assigner, build_sampler,
+                        images_to_levels, multi_apply)
+from ..builder import HEADS, build_head
+from .base_dense_head import BaseDenseHead
+from .rpn_head import RPNHead
+
+
+class AdaptiveConv(nn.Module):
+    """AdaptiveConv used to adapt the sampling location with the anchors.
+
+    Args:
+        in_channels (int): number of input channels.
+        out_channels (int): number of output channels.
+        adapt_cfg (dict): adaptation type and arguments.
+            dilation (uniform anchor) or offset (arbitrary anchors).
+    """
+
+    def __init__(self,
+                 in_channels,
+                 out_channels,
+                 adapt_cfg=dict(type='dilation', dilation=3)):
```

Comment:
Can we align the interface with other Convs?

Reply:
Do you have any suggestion to align with the interface of other Convs

Reply:
Sorry for the late reply. For example, you can follow the init method of Deform2d: https://github.com/open-mmlab/mmcv/blob/987cb58341878538a86c584757a49539bb1caab6/mmcv/ops/deform_conv.py#L181-L195.
The default value of the init method should be also the same to avoid ambiguity.
Since `AdaptiveConv` is only tested with `kernel_size=3`, we can add an assertion `assert kernel_size == 3, 'Adaptive conv only supports kernels 3'`.

Reply:
I see. Also thank you for your new commit.

Reply:
You are welcome. I am training with the latest codes, should be able to merge if the performance matches.